File: src/main/java/me/kuku/yuq/service/ConfigService.java
package me.kuku.yuq.service;

import com.IceCreamQAQ.Yu.annotation.AutoBind;
import me.kuku.yuq.entity.ConfigEntity;
import me.kuku.yuq.pojo.ConfigType;

import java.util.List;

@AutoBind
public interface ConfigService {
    List<ConfigEntity> findAll();
    void save(ConfigEntity configEntity);
    ConfigEntity findByType(String type);
    ConfigEntity findByType(ConfigType configType);
}

File: src/main/kotlin/me/kuku/yuq/service/ConfigService.kt
package me.kuku.yuq.service

import com.IceCreamQAQ.Yu.annotation.AutoBind
import me.kuku.yuq.entity.ConfigEntity

@AutoBind
interface ConfigService {
    fun findAll(): List<ConfigEntity>
    fun save(configEntity: ConfigEntity)
    fun findByType(type: String): ConfigEntity?
}